### ISSUE TITLE
slog entries missing for swing-store import / export before init

### ISSUE THREAD

**Issue by mhofman** [Agoric/Agoric#8271]
## Describe the bug

The `SWING_STORE_EXPORT` messages from cosmos can be handled by cosmic-swingset before the controller has been initialized.
Currently the handler uses `writeSlogObject` which is only assigned after init, causing no slog output detailing the processing of these messages.

## Expected behavior

Slog entries for swing-store export messages, even before init.

## Platform Environment
agoric-upgrade-11

## Additional context

`slogSender` is created by cosmic-swingset during init and wrapped by swingset into a `writeSlogObject` function. The slogSender could likely be created at program start, and the wrapping into `writeSlogObject` inverted to become a host concern. Likely related to providing a generic utility to report operation spans (see #5724)

Verify that the slog messages are handled correctly by slog-to-otel

**Comment by mhofman** [comment]
@gibson042 do you think there may be a way to create the slog sender early before we get the init message?

**Comment by gibson042** [comment]
It would take some refactoring, but... maybe?

Right now, [chain-main.js `makeLaunchChain`](https://github.com/Agoric/agoric-sdk/blob/0282f9980ab5c1f289d9fe544b7b087bae11a767/packages/cosmic-swingset/src/chain-main.js#L240) accepts a mandatory `stateDBDir` path string and an optional `env` and (via `testingOverrides`) `slogSender`, and returns a [`launchChain`](https://github.com/Agoric/agoric-sdk/blob/0282f9980ab5c1f289d9fe544b7b087bae11a767/packages/cosmic-swingset/src/chain-main.js#L301) function that requires an AG_COSMOS_INIT action from whose `resolvedConfig` it projects an optional `slogfile` (if present) into `env`, establishes `slogSender` via either `providedSlogSender` or `makeSlogSender({ stateDir: stateDBDir, env, … })`, passes `env` and `slogSender` into [launch-chain.js `launchAndShareInternals`](https://github.com/Agoric/agoric-sdk/blob/4841c8738923019bcbd3e4d41900381ba677d55f/packages/cosmic-swingset/src/launch-chain.js#L356) (usually via [`launch`](https://github.com/Agoric/agoric-sdk/blob/4841c8738923019bcbd3e4d41900381ba677d55f/packages/cosmic-swingset/src/launch-chain.js#L1459) as `launcher`) (_which passes both into [`buildSwingset`](https://github.com/Agoric/agoric-sdk/blob/4841c8738923019bcbd3e4d41900381ba677d55f/packages/cosmic-swingset/src/launch-chain.js#L161), where `slogSender` gets wrapped by a [`makeSwingsetController`](https://github.com/Agoric/agoric-sdk/blob/4841c8738923019bcbd3e4d41900381ba677d55f/packages/SwingSet/src/controller/controller.js#L138) controller's [`writeSlogObject`](https://github.com/Agoric/agoric-sdk/blob/4841c8738923019bcbd3e4d41900381ba677d55f/packages/SwingSet/src/controller/controller.js#L171)_), and returns output including a `writeSlogObject` function wrapping the controller's method. That `makeLaunchChain` is called by [chain-main.js `main`](https://github.com/Agoric/agoric-sdk/blob/0282f9980ab5c1f289d9fe544b7b087bae11a767/packages/cosmic-swingset/src/chain-main.js#L578) (_currently after `agcc.runAgCosmosDaemon(…)` but not dependent upon it_), and the `launchChain` it returns is invoked when the AG_COSMOS_INIT action actually arrives (_which, as noted above, may include the path to a slog file_).

So basically, everything necessary to construct `slogSender` is available when calling `makeLaunchChain` **except** for the slogfile path which might come from AG_COSMOS_INIT, which we'd need to address one way or another.

**Comment by mhofman** [comment]
> So basically, everything necessary to construct `slogSender` is available when calling `makeLaunchChain` **except** for the slogfile path which might come from AG_COSMOS_INIT, which we'd need to address one way or another.

Oh yeah this issue was originally created before we added that plumbing.

But that raises an interesting point. The cosmos configured options would never be able to apply to swing-store restore in that case, which is actually one of the intended use cases. See https://github.com/Agoric/agoric-sdk/issues/8629. Maybe we erred in the design of using `AG_COSMOS_INIT` to plumb the options.
